### PR TITLE
EVG-3488 support setup-failed in admin task restart

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: rest/route/admin_restart.go [hunk 1]
```diff
@@ -21,11 +21,12 @@ func makeRestartRoute(sc data.Connector, queue amboy.Queue) gimlet.RouteHandler
...
 type restartHandler struct {
-	StartTime  time.Time `json:"start_time"`
-	EndTime    time.Time `json:"end_time"`
-	DryRun     bool      `json:"dry_run"`
-	OnlyRed    bool      `json:"only_red"`
-	OnlyPurple bool      `json:"only_purple"`
+	StartTime          time.Time `json:"start_time"`
+	EndTime            time.Time `json:"end_time"`
+	DryRun             bool      `json:"dry_run"`
+	IncludeTestFailed  bool      `json:"inc_test_failed"`
```
Comment (richardsamuels) [inline rest/route/admin_restart.go]: Nit but you might as well spell out "include". I read this and though "increment? what?"

File: model/task_lifecycle.go [hunk 2]
```diff
@@ -753,22 +756,32 @@ type RestartTaskResults struct {
...
-	if opts.OnlyRed && opts.OnlyPurple {
-		opts.OnlyRed = false
-		opts.OnlyPurple = false
+	if !opts.IncludeTestFailed && !opts.IncludeSysFailed && !opts.IncludeSetupFailed {
+		opts.IncludeTestFailed = true
+		opts.IncludeSysFailed = true
+		opts.IncludeSetupFailed = true
 	}
-	tasksToRestart, err := task.Find(task.ByTimeStartedAndFailed(opts.StartTime, opts.EndTime))
+	failedTasks, err := task.Find(task.ByTimeStartedAndFailed(opts.StartTime, opts.EndTime))
```
Comment (richardsamuels) [inline model/task_lifecycle.go]: Can we just leverage the database for this operation, and build a $in query based on the options?

File: model/task_lifecycle.go [hunk 3]
```diff
@@ -732,12 +732,15 @@ func updateDisplayTask(t *task.Task) error {
...
-	OnlyPurple bool      `bson:"only_purple" json:"only_purple" yaml:"only_purple"`
-	StartTime  time.Time `bson:"start_time" json:"start_time" yaml:"start_time"`
-	EndTime    time.Time `bson:"end_time" json:"end_time" yaml:"end_time"`
-	User       string    `bson:"user" json:"user" yaml:"user"`
+	DryRun    bool      `bson:"dry_run" json:"dry_run"`
+	StartTime time.Time `bson:"start_time" json:"start_time"`
+	EndTime   time.Time `bson:"end_time" json:"end_time"`
+	User      string    `bson:"user" json:"user"`
+
+	// note that the bson tags are not quite accurate, but are kept around for backwards compatibility
```
Comment (richardsamuels) [inline model/task_lifecycle.go]: The json tags don't match what you're using in the UI.
Also I don't feel like this comment is necessary, but I don't really care if you leave it in or not.

Comment (john-m-liu) [reply model/task_lifecycle.go]: They don't, but they aren't used in the UI

File: model/task_lifecycle.go (hunk 2 above)
Comment (john-m-liu) [reply model/task_lifecycle.go]: yep, fixed

File: rest/route/admin_restart.go (hunk 1 above)
Comment (john-m-liu) [reply rest/route/admin_restart.go]: done

**Pull request merged**